origin-contracts/contracts/identity/KeyHolderLibrary.sol

```diff
@@ -73,9 +73,11 @@ library KeyHolderLibrary {
         public
         returns (bool success)
     {
-        require(_keyHolderData.keys[_key].key != _key, "Key already exists"); // Key should not already exist
```

Comment:
This means that a key can only ever have a single purpose. If that is the intention, there is no need to store purposes in an array.